bgpd/bgp_attr.c

```diff
@@ -2890,10 +2890,15 @@ bgp_packet_mpattr_start (struct stream *s, afi_t afi, safi_t safi, afi_t nh_afi,
 
 	  assert (attr->extra);
 	  bpacket_attr_vec_arr_set_vec (vecarr, BGP_ATTR_VEC_NH, s, attr);
-	  stream_putc (s, attre->mp_nexthop_len);
-	  stream_put (s, &attre->mp_nexthop_global, IPV6_MAX_BYTELEN);
-	  if (attre->mp_nexthop_len == BGP_ATTR_NHLEN_IPV6_GLOBAL_AND_LL)
+
+	  if (attre->mp_nexthop_len == BGP_ATTR_NHLEN_IPV6_GLOBAL_AND_LL) {
+	    stream_putc (s, BGP_ATTR_NHLEN_IPV6_GLOBAL_AND_LL);
+	    stream_put (s, &attre->mp_nexthop_global, IPV6_MAX_BYTELEN);
 	    stream_put (s, &attre->mp_nexthop_local, IPV6_MAX_BYTELEN);
+          } else {
+	    stream_putc (s, IPV6_MAX_BYTELEN);
```

Comment:
In this scenario for ipv4+label the mp_nexthop_len is 4 but since we are using ENHE we need to use IPV6_MAX_BYTELEN